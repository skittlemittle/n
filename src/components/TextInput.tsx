import React, { useState } from "react";
import styled from "styled-components";

/** input state handling  */
function useInput(defaultVal: string) {
  const [value, setValue] = useState(defaultVal);

  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setValue(e.target.value);
  };
  return { value, onChange };
}

const TextInput = styled.input`
  margin: 0.15em 0.25em;
  border: 1px solid ${(props) => props.theme.colors.orange};
  color: ${(props) => props.theme.colors.fg1};
  background-color: inherit;

  &:focus {
    outline: none !important;
    border-color: ${(props) => props.theme.colors.orange_l};
  }

  ::selection {
    background-color: ${(props) => props.theme.colors.bg_select};
  }
`;

export { TextInput, useInput };
