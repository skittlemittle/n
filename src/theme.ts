import { COLORS_L } from "./themeConstants";

export const AppTheme = {
  colors: COLORS_L,
};
