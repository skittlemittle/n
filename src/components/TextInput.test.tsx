import renderer from "react-test-renderer";
import { cleanup, fireEvent, render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { AppTheme } from "../theme";

import { TextInput, useInput } from "./TextInput";

it("styles", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={AppTheme}>
        <TextInput value="gaming" />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

afterEach(cleanup);

const InputWrapper = () => {
  const props = useInput("bababooey");
  return <TextInput {...props} />;
};

it("useInput", () => {
  const { queryByDisplayValue, getByDisplayValue } = render(
    <ThemeProvider theme={AppTheme}>
      <InputWrapper />
    </ThemeProvider>
  );

  expect(queryByDisplayValue(/bababooey/i)).toBeTruthy();
  fireEvent.change(getByDisplayValue(/bababooey/i), { target: { value: "g" } });

  expect(queryByDisplayValue(/g/i)).toBeTruthy();
});
