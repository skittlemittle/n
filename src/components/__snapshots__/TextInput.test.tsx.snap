// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styles 1`] = `
<input
  className="sc-bdvvtL htsTwx"
  value="gaming"
/>
`;
